### ISSUE TITLE
Ship static binaries

### ISSUE THREAD

**Issue by wucke13** [rust-analyzer/rust-analyzer#4589]
I would  like to use `rust-analyzer`. However, most ways of obtaining it require one to use a downloaded version, which I guess is build against `x86_64-unknown-linux-gnu`. This hurts the portability as these binaries will not work on distros like Alpine Linux or NixOS (at least without further ).

I would appreciate if the binaries will be build against the `x86_64-unknown-linux-musl` target in the future in order to achieve maximum compatibility. IMHO it is always wrong to ship a dynamic linked binary if static linkage is an easy option but if one exactly knows the target system (which is not the case if one binary is provided for the whole linux world).

Other than that, thank you all for the great work!

PS: For me `rust-analyzer` compiles against `x86_64-unknown-linux-musl` just fine, so this shouldn't be fairly quick fix.

**Comment by flodiebold** [comment]
Duplicate of #4354; see there and in the linked PR for why this is not possible currently.

**Comment by lnicola** [comment]
NixOS should actually work fine after #4555.

**Comment by wucke13** [comment]
Woosh, that was fast. Thanks! Though #4555 is only about VSCode, right? What about coc-rust-analyzer?

**Comment by lnicola** [comment]
`coc-rust-analyzer` is a fork of the Code extension, so @fannheyward might include the same changes. But indeed, it won't work today,

**Comment by fannheyward** [comment]
`coc-rust-analyzer` adds this patch, https://github.com/fannheyward/coc-rust-analyzer/commit/29d48111c44822ab4ced6a8ae1a18335de05d0b5, releases with 0.7.2
